index.src.html

```diff
@@ -2049,24 +2049,36 @@
 
 ### Methods ###{#audiodata-methods}
 : <dfn method for=AudioData>
-      copyTo(|destination|, |planeNumber|)
+      copyTo(|destination|, |options|)
     </dfn>
 :: Copies the samples from the specified plane of the {{AudioData}} to the
     destination buffer.
 
     When invoked, run these steps:
     1. If the value of |frame|'s {{AudioData/[[detached]]}} internal slot is
         `true`, throw an {{InvalidStateError}} {{DOMException}}.
-    2. Let |allocationSize| be the number of bytes allocated to hold this
-        {{AudioData}}'s [=media resource=], as described by the getter steps of
-        {{AudioData/allocationSize}}.
-    3. If |allocationSize| is greater than `destination.byteLength`, throw a
-        {{TypeError}}.
-    4. Let |resource| be the [=media resource=] referenced by
+    2. Let |frameCount| be the number of frames in each plane of this
+        {{AudioData}}'s' [=media resource=], as defined by the getter steps of
+        {{AudioData/numberOfFrames}}.
+    3. If |options|.{{AudioDataCopyToOptions/frameOffset}} is greater than or
+        equal to |frameCount|, throw a {{RangeError}}.
+    4. Let |copyFrameCount| be the difference of subtracting
+        |options|.{{AudioDataCopyToOptions/frameOffset}} from |frameCount|.
+    5. If |options|.{{AudioDataCopyToOptions/frameCount}} [=map/exists=]:
+        1. If |options|.{{AudioDataCopyToOptions/frameCount}} is greater than
+            |copyFrameCount|, throw a {{RangeError}}.
+        2. Otherwise, assign |options|.{{AudioDataCopyToOptions/frameCount}}
+            to |copyFrameCount|.
+    6. Let |bytesPerSample| be the number of bytes per sample, as defined by
+        the {{AudioData/[[sample format]]}}.
+    7. If the product of multiplying |bytesPerSample| by |copyFrameCount| is
+        greater than `destination.byteLength`, throw a {{RangeError}}.
```

Comment:
As said during the call, this is multiplying the sample size with a frame count, there needs to be a multiplication by the channel count if the source is interleaved. Maybe this is also an opportunity to do the conversion? We might want to do this as a separate method, not sure.

Reply:
I've changed step 2 at the top to make it such that copyFrameCount includes all channels in the case of single-plane interleaved formats. I think that means step 7 here is now correct.

I do like the idea of conversion via this method, but I'd like to send a follow up PR for that, probably after I've ironed out the AudioSampleFormat definitions as you mentioned.  One question though: if we allow conversion here, should we make the semantics of options.frameOffset and options.frameCount be post-conversion? That is, convert first, then offset into the converted buffer?

Reply:
I've changed the earlier steps at the top to make it such that, for single plane interleaved formats, this copyFrameCount actually includes frames across all channels. I think this makes the wording in step 7 correct (and reflects my original intent in this case too). LMK if it works. 

I do like the idea of conversion here, but I'd like to do that in a follow up PR. Potentially after I've expanded the definitions of AudioSampleFormats as you mentioned. One question though: if we add conversion, do we apply the options.frameOffset and options.frameCount pre or post converting? I'm thinking post, but not certain.

Reply:
> The offset and length is pre-conversion, but since it's an element count, it doesn't change post-conversion.

Yes, should be true now.

> If one wants to place frames at a specific offset in an existing buffer this can be done using a typed array view, no ?

I agree